### ISSUE TITLE
Please adjust the new CPLD Adapter for the A1000

### ISSUE THREAD

**Issue by SirCathal** [LinuxJedi/AmigaRGBtoHDMI#5]
Hello, I'm really looking forward to the CPLD version for the A500. I've already ordered components.
But would it be possible that you adapt the A500 version so that it fits into the A1000.
In the A1000, the Denise chip is rotated 180 ° compared to the A500. That means with the new adapter the Pi would hang out of the front of the case. So it doesn't fit in.

It would look like this:
![WrongSidemarked](https://user-images.githubusercontent.com/69156692/114773691-a62f9900-9d6f-11eb-95ca-0507a1a3c4ef.jpg)

Based on your new adapter, the Pi would simply have to be on the other side, not to the right but to the left (see photo for explanation).
So it could work so that neither the front nor the power supply are in the way.
![PossibleMarked](https://user-images.githubusercontent.com/69156692/114773799-c95a4880-9d6f-11eb-896b-3ae68d75d43a.jpg)

Since the A1000 always has flickering pixels, because the timing is just a little different there than in the newer Amigas, your solution would be perfect for the A1000.

@LinuxJedi 
It would be totally great if you could adjust that accordingly. Unfortunately, I can't design circuit boards, but if you can do that I would test it right away.

**Comment by LinuxJedi** [comment]
Sure, I won't be able to until this weekend at the earliest, but should only take an hour or so.

**Comment by SirCathal** [comment]
That would be great! Time is not a problem. Here in Germany I have to wait a week or two for the chips that I ordered in the US to arrive anyway.
Here they were unfortunately not in stock or significantly more expensive.
Thank you very much in advance!

**Comment by LinuxJedi** [comment]
@SirCathal I've just merged #6 which might do the trick. Let me know what you think.

**Comment by rlake** [comment]
Waiting on boards to arrive but can give this a go on an NTSC board assuming a faster (5 or 7) CPLD is fine.

**Comment by LinuxJedi** [comment]
> Waiting on boards to arrive but can give this a go on an NTSC board assuming a faster (5 or 7) CPLD is fine.

Yes, a faster CPLD will work but is more expensive, Mouser still have stock if that helps: https://www.mouser.co.uk/ProductDetail/Xilinx/XC9572XL-10VQ44C

I should note that you will need to modify your A1000 board to add CSYNC to the Denise socket. I'll soon do an updated revision soon to optionally hook a wire elsewhere to provide that. @SirCathal has details of the change that is required.

**Comment by rlake** [comment]
Sounds good, I somehow missed the non-RoHS version but no matter.

I'll try a clip onto pin 2 of U7A/MC1377 for CSYNC.

**Comment by SirCathal** [comment]
In My PAL A1000, I soldered a cable from pin 8 on UA6 to pin 32 of the Denise socket under the Moterboard to get CSYNC.
Here is the fix from below the mainboard.
(1) is Pin 8 of UA6
(2) is Pin 32 of the Denise socket.
![CSYNC below](https://user-images.githubusercontent.com/69156692/116982360-f5ab1a00-acc8-11eb-90cd-1b88a199b3aa.jpg)

@LinuxJedi all parts for your A1000 adapter arrived. A friend soldered it for me because I'm not so good in soldering SMD stuff. 
He already tested it with his A500 board (he don't have a A1000). There it worked very good. Now he sanded it to me and it will arrive this week. Then I will do the final Test in my A1000.
If it's done I will post Pictures and a Video.

**Comment by LinuxJedi** [comment]
Perfect, thank you. When I have confirmation it is good I'll update the files to add a place to hook a wire into my board instead and write up some documentation. Can you please take a photo of the top of UA6 for me? It will help me mark in the documentation what to do.

**Comment by SirCathal** [comment]
No Problem, here is the Foto. Please make sure that this is for the PAL Board with no PiggyBoard.
As far as I Know, the NTSC Board has the signal somewhere else.
![CSYNC Top](https://user-images.githubusercontent.com/69156692/117034405-91597c00-ad03-11eb-95df-2531265d6712.jpg)

**Comment by SirCathal** [comment]
I Hope I can do test final test at the weekend. That the parcel did not move today, I think DHL hates me....

**Comment by SirCathal** [comment]
I Hope I can do test final test at the weekend. That the parcel did not move today, I think DHL hates me....

**Comment by SirCathal** [comment]
@LinuxJedi 
I can confirm, The A1000 Adapter works and fits perfectly in my Pal A1000 with CSYNC added as in my description above. 
Thank you for this wonderful peace of hardware.

**Comment by LinuxJedi** [comment]
> @LinuxJedi
> I can confirm, The A1000 Adapter works and fits perfectly in my Pal A1000 with CSYNC added as in my description above.
> Thank you for this wonderful peace of hardware.

Excellent! Glad to hear it! Can I be cheeky and ask for a photo of the motherboard with it in please? I want to see how close my estimates based on photos were :)

**Comment by SirCathal** [comment]
No Problem, here the Pictures.
Ignore the black wire, it's only a ground wire to the Pi0 from old experiments.

![A1000 Adapter with Pi0](https://user-images.githubusercontent.com/69156692/117547943-86b82300-b032-11eb-999b-341f26614442.jpg)
![A1000 Adapter](https://user-images.githubusercontent.com/69156692/117547947-8b7cd700-b032-11eb-8522-1483ad24d1e1.jpg)

**Comment by LinuxJedi** [comment]
Awesome! Thanks so much. At some point this weekend I'll write-up some documentation, revise slightly to add an optional pin to attach CSYNC to and close this ticket.

**Comment by rlake** [comment]
Just trying this out now. I used a JTAG cable to program the chip through GPIO, extracted RGBtoHDMI_20210322_f771e51 and changed single_button_mode=1 in Default.txt

Booting gives the CPLD Recovery Menu, which sounds wrong unless I missed some instructions. Any suggestion for where to check next? I also need to learn how to use the menu system

**Comment by rlake** [comment]
Found my serial cable and well, it doesn't look good for my build for some reason! I'll have to investigate more into how the detection works before I try reflowing and/or programming again:

```
Unknown CPLD: identifier = fff
EMMC: BCM2708 controller power-cycled
Keycount = 0
CPLD  Design: Unknown
CPLD Version: f.f
```

Anyway, here's the fit for an NTSC Amiga 1000. It's a little snug with that brown "DL1" component but it's not too bad with the "piggy board". For CSYNC I'm just using a test clip from pin 8 of U6A (CSY* according to my Zorro schematics) to a wire fed under the socket to Denise pin 32.

I suppose if there's something to wish for on a final rev it'd be some optional headers for 3-button mode and the uart tx+gnd just in case! 😀

![RikkiAmigaRGBToHDMI1000Fit](https://user-images.githubusercontent.com/372855/117558417-1e694180-b032-11eb-8658-22b94ddcf20a.jpg)

**Comment by LinuxJedi** [comment]
> Just trying this out now. I used a JTAG cable to program the chip through GPIO, extracted RGBtoHDMI_20210322_f771e51 and changed single_button_mode=1 in Default.txt
> 
> Booting gives the CPLD Recovery Menu, which sounds wrong unless I missed some instructions. Any suggestion for where to check next? I also need to learn how to use the menu system

The recovery menu doesn't work in single_button_mode in that version unfortunately, there is a bug which has been fixed but not made it into the main releases yet.

I highly recommend using the latest beta from here: https://github.com/IanSB/RGBtoHDMI/releases

Then you can set single_button_mode=1 in Default.txt and use the recovery menu properly to flash the board. It also has some NTSC specific fixes in it.

**Comment by rlake** [comment]
Yeah that helped a lot! This time I generated an XSVF from Xilinx tools and dropped it into the 6-12 bit and recovery folders, but the menu didn't display the file (not sure how that's determined) so I went with the BBC firmware. RPi programmed fine:

```
CPLD  Design: 6-12_BIT_RGB
CPLD Version: 9.2
```

But that leaves me with a grey screen and no means to get back into recovery menu (to my knowledge, button doesn't have any effect; serial console shows it booting). I can't explain why the Xilinx tools with my Digilent/Adept adapter/plugin "verified" the firmware at first (and didn't detect) but the RPi programmed firmware is good.

I'll try and dig through the software later to see what's going on with the menu, recovery, why it's in "Acorn" mode etc. Hardware seems good so far though, stuck with software/tools debugging. :)

**Comment by LinuxJedi** [comment]
Latest commit to main adds two more optional buttons and an optional pin to hook CSYNC into and a README is added. rlake's issues are being diagnosed elsewhere. I'm going to close this for now. Please feel free to open a new issue with any issues found.
